k2/csrc/array.h

```diff
@@ -391,17 +389,33 @@ class Array2 {
         elem_stride0_(elem_stride0),
         dim1_(dim1),
         byte_offset_(byte_offset),
-        region_(region) {}
+        region_(region) {
+    K2_CHECK_GE(dim0_, 0);
+    K2_CHECK_GE(dim1_, 0);
+    K2_CHECK_GE(elem_stride0_, dim1_);
+    K2_CHECK_GE(byte_offset_, 0);
+  }
+
+  Array2 ToContiguous() const {
+    if (elem_stride0_ == dim1_) return *this;
+    Array2 ans(Context(), dim0_, dim1_);
+    auto *dst = ans.Data();
+    const auto *src = Data();
+    auto dim1 = dim1_;
```

Comment:
please don't use `auto` except for complex types, it makes the code less easy to understand.

Reply:
Fixed.